httpclient-jetty/src/main/java/io/fabric8/kubernetes/client/jetty/JettyHttpClient.java

```diff
@@ -0,0 +1,198 @@
+import io.fabric8.kubernetes.client.http.HttpResponse;
+import io.fabric8.kubernetes.client.http.Interceptor;
+import io.fabric8.kubernetes.client.http.StandardHttpRequest;
+import io.fabric8.kubernetes.client.http.WebSocket;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.client.util.BufferingResponseListener;
+import org.eclipse.jetty.client.util.InputStreamRequestContent;
+import org.eclipse.jetty.client.util.StringRequestContent;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import static io.fabric8.kubernetes.client.http.StandardMediaTypes.APPLICATION_OCTET_STREAM;
+import static io.fabric8.kubernetes.client.http.StandardMediaTypes.TEXT_PLAIN;
+import static org.eclipse.jetty.util.BufferUtil.toArray;
+
+public class JettyHttpClient implements io.fabric8.kubernetes.client.http.HttpClient {
+
+  private final HttpClient jetty;
+  private final WebSocketClient jettyWs;
+  private final Collection<Interceptor> interceptors;
+  private final JettyHttpClientBuilder builder;
+  private final JettyHttpClientFactory factory;
+
+  public JettyHttpClient(JettyHttpClientBuilder builder, HttpClient httpClient, WebSocketClient webSocketClient,
+      Collection<Interceptor> interceptors, JettyHttpClientFactory jettyHttpClientFactory) {
+    this.builder = builder;
+    this.jetty = httpClient;
+    this.jettyWs = webSocketClient;
+    this.interceptors = interceptors;
+    this.factory = jettyHttpClientFactory;
+  }
+
+  @Override
+  public void close() {
+    try {
+      jetty.stop();
+      jettyWs.stop();
+    } catch (Exception e) {
+      throw KubernetesClientException.launderThrowable(e);
+    }
+  }
+
+  @Override
+  public DerivedClientBuilder newBuilder() {
+    return builder.copy();
+  }
+
+  @Override
+  public <T> CompletableFuture<HttpResponse<T>> sendAsync(HttpRequest originalRequest, Class<T> type) {
+    final var supportedResponse = JettyHttpResponse.SupportedResponse.from(type);
+    final var request = toStandardHttpRequest(originalRequest);
+    final CompletableFuture<HttpResponse<T>> future = new CompletableFuture<>();
+    newRequest(request).send(new BufferingResponseListener() {
+
+      // TODO: Long Term Refactor - This Listener blocks until the full response is read and keeps it in memory.
+      //       Find a way to stream the response body without completing the future
+      //       We need two signals, one when the response is received, and one when the body is completely
+      //       read.
+      //       Should this method be completely replaced by consumeXxx()?
```

Comment:
There's two thing here:
- First one is addressing the in-memory load of the response, which I'll fix in a subsequent PR. 
- Second is about "merging" the `sendAsync` and `consumeLines`/`consumeBytes` since they seem redundant. I guess aligned with your thoughts. An issue for this sounds good.

Reply:
https://github.com/fabric8io/kubernetes-client/issues/4201